backend/src/test/java/io/github/projektalmanac/amoxtli/backend/service/UserServiceTest.java

```diff
@@ -461,4 +499,114 @@ void getLibrosUsuario() {
         Assertions.assertEquals(volumeInfo.getTitle(), result.getLibros().get(0).getTitulo());
         Assertions.assertEquals(URI.create(imageLinks.getMedium()), result.getLibros().get(0).getUrlPortada());
     }
-}
\ No newline at end of file
+
+    @Test
+    void getIntercambios(){
+
+        //Caso 1: El usuario no existe
+        when(userRepository.findById(3)).thenThrow(new UserNotFoundException(3));
```

Comment:
```suggestion
        when(userRepository.findById(3)).thenReturn(Optional.empty());
```

No se supone que los repositorios regresan `null` o `Option.empty()` cuando no encuentran lo que se les pide?